src/main/java/com/code_intelligence/jazzer/instrumentor/HookMethodVisitor.kt

```diff
@@ -92,6 +151,36 @@ private class HookMethodVisitor(
             return
         }
 
+        val skipHooksLabel = Label()
+        val applyHooksLabel = Label()
+        val useConditionalHooks = classWithHooksEnabledField != null
+        var postCallFrame: Pair<Array<Any>?, Array<Any>?>? = null
+        if (useConditionalHooks) {
+            val preCallFrame = (mv as? AnalyzerAdapter)?.let { storeFrame(it) }
+            // If hooks aren't enabled, skip the hook invocations.
+            mv.visitFieldInsn(
+                Opcodes.GETSTATIC,
+                classWithHooksEnabledField,
+                "hooksEnabled",
+                "Z",
+            )
+            mv.visitJumpInsn(Opcodes.IFNE, applyHooksLabel)
+            mv.visitMethodInsn(opcode, owner, methodName, methodDescriptor, isInterface)
+            postCallFrame = (mv as? AnalyzerAdapter)?.let { storeFrame(it) }
+            mv.visitJumpInsn(Opcodes.GOTO, skipHooksLabel)
+            if (preCallFrame != null) {
+                mv.visitFrame(
+                    Opcodes.F_NEW,
+                    preCallFrame.first?.size ?: 0,
+                    preCallFrame.first,
+                    preCallFrame.second?.size ?: 0,
+                    preCallFrame.second,
+                )
+            }
+            mv.visitLabel(applyHooksLabel)
+            mv.visitInsn(Opcodes.NOP)
```

Comment:
I don't think it is needed, but for a different reason: The `AnalyzerAdapter` gets confused on duplicate frames, so we need the `NOP` in the other location since we don't control the instruction that follows and it may have a stack map frame attached. We know that this isn't the case here. I added a comment explaining this.